src/pages/index.js

```diff
@@ -7,96 +7,208 @@ import Card from "../components/Card.js";
+/* ---------------------------------------------------------------------------------------------- */
 
+//! Do Not Delete
 function openProfilePopup() {
   const { profileName, description } = userInfo.getUserInfo();
   profileTitleElement.value = profileName;
   profileDescriptionElement.value = description;
   editProfileFormValidator.toggleButtonState();
   editProfilePopup.open();
+  editProfilePopup.setLoading(false, "Save");
 }
 
-/* ---------------------------------------------------------------------------------------------- */
-/*                                          Section Class                                         */
-/* ---------------------------------------------------------------------------------------------- */
-
-const cardListSection = new Section(
-  {
-    items: initialCards,
-    renderer: ({ name, link }) => {
-      const newCard = createCard({ name, link });
-      cardListSection.addItem(newCard);
-    },
-  },
-  cardList
-);
-
-cardListSection.renderItems();
+//! Do Not Delete
+function handleProfileFormSubmit({ title, description }) {
+  api
+    .updateUserInfo(title, description)
+    .then(() => {
+      userInfo.setUserInfo(title, description);
+      editProfilePopup.setLoading(true);
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
 
 /* ---------------------------------------------------------------------------------------------- */
-/*                                      PopupWithForm Classes                                     */
+/*                                   Profile Set Event Listeners                                  */
 /* ---------------------------------------------------------------------------------------------- */
 
-const newCardPopup = new PopupWithForm(cardModalSelector, submitCard);
-const previewImagePopup = new PopupWithImage(previewImageModal);
+profileEditButton.addEventListener("click", openProfilePopup);
 
-addNewCardButton.addEventListener("click", () => {
-  addCardFormValidator.toggleButtonState();
-  newCardPopup.open();
+profileAvatarPeniclIcon.addEventListener("click", () => {
+  avatarFromValidator.toggleButtonState();
+  editAvatarPopup.open();
+  editAvatarPopup.setLoading(false, "Save");
 });
 
 /* ---------------------------------------------------------------------------------------------- */
 /*                                         Card Functions                                         */
 /* ---------------------------------------------------------------------------------------------- */
 
-function createCard({ name, link }) {
-  const cardElement = new Card({ name, link }, "#card-template", ({ name, link }) => {
-    previewImagePopup.open({ name, link });
-  });
-  return cardElement.generateCard();
+//! Do Not Delete
+function handleProfileAvatarSubmit(url) {
+  api
+    .setUserAvatar(url)
+    .then((userData) => {
+      editAvatarPopup.setLoading(true);
```

Comment:
`.setLoading(true)` should be called **before** sending requests to see the button text changing